Homework/text_box_page.py

```diff
@@ -0,0 +1,28 @@
+from selenium.webdriver.common.by import By
+
+
+class TextBoxPage:
+    def __init__(self, driver):
+        self.driver = driver
+
+    def enter_full_name(self, full_name):
+        self.driver.find_element(By.ID, "userName").send_keys(full_name)
+
+    def enter_email(self, email):
+        self.driver.find_element(By.ID, "userEmail").send_keys(email)
+
+    def enter_current_address(self, current_address):
+        self.driver.find_element(By.ID, "currentAddress").send_keys(current_address)
+
+    def enter_permanent_address(self, permanent_address):
+        self.driver.find_element(By.ID, "permanentAddress").send_keys(permanent_address)
+
+    def click_submit(self):
+        self.driver.find_element(By.ID, "submit").click()
```

Comment:
The test is failing to submit the button.
```plaintext
FAILED test_form.py::test_submit_form - selenium.common.exceptions.ElementClickInterceptedException: Message: element click intercepted: Element <button id="submit" type="button" class="btn btn-primary">...</button> is not clickable at point (1078, 662).
```

This is because, during the execution, the "Submit" button is not visible to the driver. To solve it, you should do a small scroll down to the driver to be able to find it.

```python
    def click_submit(self):
        submit_button = self.driver.find_element(By.ID, "submit")
        self.driver.execute_script("arguments[0].scrollIntoView();", submit_button)
        submit_button.click()
```